cpp/references/data_structures/list_digraph.hpp

```diff
@@ -0,0 +1,230 @@
+#ifndef LIST_DIGRAPH_HPP
+#define LIST_DIGRAPH_HPP
+
+#include <Godot.hpp>
+#include <Reference.hpp>
+
+#include <lemon/list_graph.h>
+#include <lemon/concepts/maps.h>
+
+namespace godot {
+
+#define _STR(m_x) #m_x
+
+#define ERR_FAIL_PARAM(m_param, m_type)																						\
+{																															\
+	Godot::print_error("Error: '" _STR(m_param) "' is not of type " _STR(m_type) ".", __FUNCTION__, __FILE__, __LINE__);	\
+}																															\
+
+#define ERR_NULL_PARAM(m_param)																								\
+{																															\
+	Godot::print_error("Error: '" _STR(m_param) "' is null or of invalid type.", __FUNCTION__, __FILE__, __LINE__);			\
+}																															\
+
+class IDataOwner {
```

Comment:
Interfaces should have their own file, otherwise it defeats the point of having them since you have to include both the interface *and* the dependency using `#include "filepath"`.